### ISSUE TITLE
Occupancy sensor not updated when timer finishes

### ISSUE THREAD

**Issue by wimleers** [teh-hippo/homebridge-eggtimer-plugin#108]
Hi! Been using this plugin with great success for months, until I started using it in a slightly more advanced way yesterday.

When the occupancy sensor is enabled, the following happens:
![IMG_1009](https://github.com/teh-hippo/homebridge-eggtimer-plugin/assets/14240/c17a59f1-0bc2-41a0-a79f-f416cd3b1301)

As you can see, it nicely counts down from 3% to 2% after 1 minute, etc., until it reaches 0% and "turns off" after 3 minutes. Great 👍

But … the occupancy sensor is _not_ updated!

That seems to be happening because https://github.com/teh-hippo/homebridge-eggtimer-plugin/commit/c8c451396f379c0e9492e789ddd93baaed4bf593 does not _automatically_ update the state of the occupancy sensor; it only computes a new value if it is _requested_.

Which means that my use of `OccupancyDetected===No` as a trigger **fails 😞 **

The fix is trivial fortunately 😄

**Comment by wimleers** [comment]
1. https://controllerforhomekit.com/ 😊 (I prefer the UX of the Eve app for inspecting characteristics, that of Home+ for creating automations/doing something quickly and Controller for debugging problems/finetuning automations and scenes to make them consistent)
2. I use multiple Aqara FP2 mmwave occupancy sensors in my apartment. To turn on the kitchen and dining table lights and avoid them turning on/off very quickly, I use your plugin to set a minimum duration, to avoid flicker (FP2 detects absence → light would turn off within a second, I'd return a few seconds later, it'd turn on again — annoying!). 
![IMG_1012](https://github.com/teh-hippo/homebridge-eggtimer-plugin/assets/14240/b9fe27a3-cca5-46f8-8762-f928d7fec7af)
![IMG_1013](https://github.com/teh-hippo/homebridge-eggtimer-plugin/assets/14240/242e12f3-dfb0-47eb-b86f-98c7d1bac64f)

**Comment by wimleers** [comment]
> I've made the fix, but haven't tested it. I'll test it tonight and should get that fixed.

FYI: happy to test it myself too. I maintain a Homebridge plugin and contribute to another, so I'm familiar with the `npm link` dance to test a PR/commit 🤓

**Comment by wimleers** [comment]
> Negative though is that I don't know how many calls I can make to set the brightness in automation rules before its throttled in a very classic Apple way. E.g., For one complex rule, I set it to 90% to indicate its been triggered, 80% for path A, 70% for path B etc. It hasn't stayed at 90% yet... but never know.

I use different brightnesses as a condition in dozens of automations for that same reason. I wouldn't worry about it — it works fine 😊  (I have 99 scenes, 179 automations and 127 accessories right now.)

**Comment by wimleers** [comment]
> I see you've got non-standard characteristics. Will have a read through your approach.

I highly recommend going this way. All these "brightness actually means this other thing" things are super confusing. The only benefit is you can then set up the automations in the Home app. But the Home app doesn't even allow for even slightly advanced automations, so what's the point?

I'd rather be forced to use Eve/Controller/Home+ and have sensible information than having to remember what "brightness" (or "light level" or …) means for every accessory 😅 

I hope the screenshots on my project page speak for themselves 🤞 

P.S.: chances are you can reuse the modbus stuff in my plugin — some of that's supported by most inverters. You'll need to rip out the energy manager stuff though.

**Comment by wimleers** [comment]
Oh and … thanks for the very fast fix + release! 🤩 👏

**Comment by wimleers** [comment]
Updated to 1.0.19 and … now it seems to have the opposite problem: it sets `Power=off` immediately as well as `OcccupancyDetected=no` immediately? 😮 

Seems like there's some kind of race condition?

**Comment by wimleers** [comment]
> The occupancy sensor should now be updated specifically when the timer concludes.

Yes, it should. But it doesn't. 😬 

The code looks correct to me too.

Which is why I said I suspect a non-obvious race condition to be causing that.

When I say "immediately", I mean _immediately_, meaning BEFORE the brightness reaches 0%! 😨

**Comment by wimleers** [comment]
Okay I'm pretty sure now there is a race condition happening here (in version 1.0.19), because I've now had this happen _twice_:

![IMG_1022](https://github.com/teh-hippo/homebridge-eggtimer-plugin/assets/14240/010ac7b8-606c-4474-84c1-76c2ed040500)
![IMG_1024](https://github.com/teh-hippo/homebridge-eggtimer-plugin/assets/14240/f37d70b4-f535-4db3-b801-052241666b16)

**Comment by wimleers** [comment]
… but when that ☝️ does not happen, _then_ it seems to work fine:
![IMG_1023](https://github.com/teh-hippo/homebridge-eggtimer-plugin/assets/14240/4f483e29-6dbd-4cf1-8a1a-20f677611635)

**Comment by wimleers** [comment]
It sounds very plausible that that would have fixed it! 👏

I'm trying to simplify my Homebridge setup, and because I already am using https://github.com/kernie66/homebridge-random-delay-switches for its cron-style switches (to work around iOS 17 dropping support for "<something> changes to ANY VALUE", i.e. by checking a combination of states every 2 minutes, cron-style), I realized I could achieve the same there. So I started using that. And it works well.

One fewer plugin … but I definitely do not like the terrible configuration UX of that plugin, plus the gazillion useless (to me) characteristics it exposes. Still, sticking with fewer plugins … for now.

Thanks for your excellent work, I'm looking forward to finding a use case for your plugin again some time 😊
